### PR TITLE
Adding Migration Guide 'Overview'

### PR DESCRIPTION
**Pull request by SharpRake**
## Type of change

This PR adds a customer migration doc to Academy.

This doc is intended to be something of an introduction to migrating to Chainguard Images and mostly consists of links to existing resources. 

### What should this PR do?

Resolves https://github.com/chainguard-dev/customer-issues/issues/1575

### Why are we making this change?

Customers don't know where to get started migrating to Chainguard Images, hopefully this helps.

### What are the acceptance criteria? 

Sales folks (hey @metalstormbass!) gave their feedback last week.

### How should this PR be tested?

Just a close read, no tech test to speak of.

### CODE REVIEW

File: content/chainguard/migration/migrations-overview.md [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+1 2 3 4 5
+
+423450e3fd52:/#
+```
+
+#### Use `apk search`
+
+Following on from the last point, you'll often need to install extra utilities to provide required dependencies for applications and scripts. These dependencies are likely to have different package names compared to other Linux distributions, so the apk search command can be very useful for finding the package you need. 
+
+For example, say we are porting a Dockerfile that uses the groupadd command. We could convert this to the busybox addgroup equivalent, but it's also perfectly fine to add the groupadd utility. The only issue is that there's no groupadd package, so we have to search for it:
```
Comment (amouat) [inline content/chainguard/migration/migrations-overview.md:118]: could put commands like addgroup in the `s

**Review by amouat — APPROVED** [review]
LGTM, few minor things that could be fixed.

File: content/chainguard/migration/migrations-overview.md [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+* Chainguard Images typically don't run as root, so a `USER root` statement may be required before installing software.
+* The `-dev` images and `wolfi-base` use BusyBox by default, so any `groupadd` or `useradd` commands will need to be ported to `addgroup` and `adduser`.
+* The free Developer tier of Images provides `:latest` and `:latest-dev` versions. Our paid Production Images offer tags or older versions.
+* We use apk tooling, so `apt get` commands will become `apk add`.
+* Chainguard Images are based on `glibc` and our packages cannot be mixed with Alpine packages.
+* The entrypoint on Chainguard Images is likely to be different from other common images (due to the lack of a shell) which can be confusing; for example, shell commands may get unexpectedly run by the Python interpreter.
+* In general, Chainguard recommends using `chainguard-base` or a `-dev` image to install an application's OS-level dependencies.
```
[the remark this answers is not in the document]
Comment (SharpRake) [reply content/chainguard/migration/migrations-overview.md:43]: Sales specifically asked me to cut references to wolfi in this doc - there's a worry that when we talk about Wolfi it confuses folks. 

I'll make this a bit more generalized by saying "recommends using a base image like `chianguard-base` or a `-dev` image...."

File: content/chainguard/migration/migrations-overview.md [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+Because of their minimalist design, Chainguard Images sometimes require users to adjust their image workflows. This document is intended to serve as a migration guide for customers transitioning their organizations to use Chainguard Images. It includes general tips and strategies for migrating to Chainguard Images as well as a curated set of migration-related resources.
+
+## Migrating to Chainguard Images
+
+### TL;DR: Porting Key Points
+
+* Chainguard's distroless Images have no shell or package manager by default. This is great for security, but sometimes you need these things, especially in builder images. For those cases we have `-dev` images (such as `cgr.dev/chainguard/python:latest-dev`) which do include a shell and package manager.
+* Chainguard Images typically don't run as root, so a `USER root` statement may be required before installing software.
+* The `-dev` images and `wolfi-base` use BusyBox by default, so any `groupadd` or `useradd` commands will need to be ported to `addgroup` and `adduser`.
+* The free Developer tier of Images provides `:latest` and `:latest-dev` versions. Our paid Production Images offer tags or older versions.
```
[the remark this answers is not in the document]
Comment (SharpRake) [reply content/chainguard/migration/migrations-overview.md:39]: oh wow, nice catch

File: content/chainguard/migration/migrations-overview.md [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+tags: ["Images", "Product", "Procedural"]
+images: []
+menu:
+  docs:
+    parent: "migration"
+weight: 005
+toc: true
+---
+
+[Chainguard Images](https://www.chainguard.dev/chainguard-images?utm_source=docs) are a collection of container images designed for security and minimalism. Many Chainguard Images are [distroless](/chainguard/chainguard-images/getting-started-distroless/); they contain only an open-source application and its runtime dependencies. These images do not even contain a shell or package manager, because the fewer dependencies a given piece of software uses, the lower likelihood that it will be impacted by CVEs.
```
[the remark this answers is not in the document]
Comment (SharpRake) [reply content/chainguard/migration/migrations-overview.md:18]: works for me!

File: content/chainguard/migration/migrations-overview.md [hunk 5]
```diff
@@ -0,0 +1,305 @@
+---
+title: "Migrating to Chainguard Images"
+linktitle: "Overview"
```
[the remark this answers is not in the document]
Comment (SharpRake) [reply content/chainguard/migration/migrations-overview.md:3]: I'll add "Overview of" to differentiate a bit

File: content/chainguard/migration/migrations-overview.md [hunk 6]
```diff
@@ -0,0 +1,305 @@
...
+#### If necessary, install a different shell
+
+The `-dev` images and `wolfi-base` images use the [ash](https://en.wikipedia.org/wiki/Almquist_shell) shell from BusyBox by default. This is nice from a minimalism perspective, but it's not so great if you need to port a bash and Debian centric entrypoint script to Chainguard Images.
+
+In these cases you have a choice — you can persevere and update your scripts to work in ash, or you can install the shell that works with your scripts. There's no reason to be stuck on the ash shell if you really need bash or zsh.
+
+For example:
+
+```Example
+docker run -it cgr.dev/chainguard/chainguard-base
```
[the remark this answers is not in the document]
Comment (SharpRake) [reply content/chainguard/migration/migrations-overview.md:94]: again, this was a request from sales. I'll add a note that this example uses a production-only image

**Comment by smythp** [comment]
@SharpRake  I think this PR introduced a name collision, there's a warning now in Hugo builds. Kind of annoying since we like to have multiple pages named overview, etc. Can be resolved by adding whitespace after the left navbar name for the overview, I've done it before with similar collisions.

**Comment by SharpRake** [comment]
Noted, thanks @smythp. I'll put up another commit soon, though it may be in a day or two (i've already got an update to this doc in the works)

**Pull request merged**